### PR TITLE
Grab updates from latest vendored changes

### LINKED ISSUE #3214: issue with slashes in branch name? or issue with pip install from branch
Be sure to check the existing issues (both open and closed!), and make sure you are running the latest version of Pipenv.

Check the [diagnose documentation](https://docs.pipenv.org/diagnose/) for common issues before posting! We may close your issue if it is very similar to one of them. Please be considerate, or be on your way.

Make sure to mention your debugging experience if the documented solution failed.

### Issue description

Looks like pipenv is erroring out when installing a branch, perhaps has to do with the slash in the name: `fatal: repository 'https://github.com/sarugaku/pythonfinder.git@feature/asdf-support/' not found`

### Expected result

I expect it to install properly.

### Actual result

When possible, provide the verbose output (`--verbose`), especially for locking and dependencies resolving issues.

### Steps to replicate

Provide the steps to replicate (which usually at least includes the commands and the Pipfile).

```
mkdir newproject
cd newproject
pipenv install --python=python # get around some pythonfinder issues for now on my laptop
pipenv install -e git+https://github.com/sarugaku/pythonfinder.git@feature/asdf-support#egg=pythonfinder
```

```
$ pipenv install -e git+https://github.com/sarugaku/pythonfinder.git@feature/asdf-support#egg=pythonfinder
Installing -e git+https://github.com/sarugaku/pythonfinder.git@feature/asdf-support#egg=pythonfinder…
Obtaining pythonfinder from git+https://github.com/sarugaku/pythonfinder.git@feature/asdf-support#egg=pythonfinder
  Cloning https://github.com/sarugaku/pythonfinder.git (to revision feature/asdf-support) to /Users/johria/.local/share/virtualenvs/newproject-pYEZLETE/src/pythonfinder
Switched to a new branch 'feature/asdf-support'
Branch 'feature/asdf-support' set up to track remote branch 'feature/asdf-support' from 'origin'.
  Installing build dependencies: started
  Installing build dependencies: finished with status 'done'
Collecting attrs (from pythonfinder)
  Using cached https://files.pythonhosted.org/packages/3a/e1/5f9023cc983f1a628a8c2fd051ad19e76ff7b142a0faf329336f9a62a514/attrs-18.2.0-py2.py3-none-any.whl
Collecting cached-property (from pythonfinder)
  Using cached https://files.pythonhosted.org/packages/3b/86/85c1be2e8db9e13ef9a350aecd6dea292bd612fa288c2f40d035bb750ded/cached_property-1.5.1-py2.py3-none-any.whl
Collecting click (from pythonfinder)
  Using cached https://files.pythonhosted.org/packages/fa/37/45185cb5abbc30d7257104c434fe0b07e5a195a6847506c074527aa599ec/Click-7.0-py2.py3-none-any.whl
Collecting crayons (from pythonfinder)
Collecting packaging (from pythonfinder)
  Using cached https://files.pythonhosted.org/packages/89/d1/92e6df2e503a69df9faab187c684585f0136662c12bb1f36901d426f3fab/packaging-18.0-py2.py3-none-any.whl
Collecting six (from pythonfinder)
  Using cached https://files.pythonhosted.org/packages/67/4b/141a581104b1f6397bfa78ac9d43d8ad29a7ca43ea90a2d863fe3056e86a/six-1.11.0-py2.py3-none-any.whl
Collecting vistir[spinner]>0.2.0 (from pythonfinder)
  Using cached https://files.pythonhosted.org/packages/3a/44/a7e8cc7eb8ba7d06940a1cc31b2a2bf091607b4fd70ae931348ece193297/vistir-0.2.4-py2.py3-none-any.whl
Collecting colorama (from crayons->pythonfinder)
  Using cached https://files.pythonhosted.org/packages/0a/93/6e8289231675d561d476d656c2ee3a868c1cca207e16c118d4503b25e2bf/colorama-0.4.0-py2.py3-none-any.whl
Collecting pyparsing>=2.0.2 (from packaging->pythonfinder)
  Using cached https://files.pythonhosted.org/packages/71/e8/6777f6624681c8b9701a8a0a5654f3eb56919a01a78e12bf3c73f5a3c714/pyparsing-2.3.0-py2.py3-none-any.whl
Collecting requests (from vistir[spinner]>0.2.0->pythonfinder)
  Using cached https://files.pythonhosted.org/packages/ff/17/5cbb026005115301a8fb2f9b0e3e8d32313142fe8b617070e7baad20554f/requests-2.20.1-py2.py3-none-any.whl
Collecting cursor; extra == "spinner" (from vistir[spinner]>0.2.0->pythonfinder)
Collecting yaspin; extra == "spinner" (from vistir[spinner]>0.2.0->pythonfinder)
  Using cached https://files.pythonhosted.org/packages/13/0d/d5f80c462f4c00d222616ddab37901ed4a587de5c3f1407a909ea58daf38/yaspin-0.14.0-py2.py3-none-any.whl
Collecting idna<2.8,>=2.5 (from requests->vistir[spinner]>0.2.0->pythonfinder)
  Using cached https://files.pythonhosted.org/packages/4b/2a/0276479a4b3caeb8a8c1af2f8e4355746a97fab05a372e4a2c6a6b876165/idna-2.7-py2.py3-none-any.whl
Collecting chardet<3.1.0,>=3.0.2 (from requests->vistir[spinner]>0.2.0->pythonfinder)
  Using cached https://files.pythonhosted.org/packages/bc/a9/01ffebfb562e4274b6487b4bb1ddec7ca55ec7510b22e4c51f14098443b8/chardet-3.0.4-py2.py3-none-any.whl
Collecting urllib3<1.25,>=1.21.1 (from requests->vistir[spinner]>0.2.0->pythonfinder)
  Using cached https://files.pythonhosted.org/packages/62/00/ee1d7de624db8ba7090d1226aebefab96a2c71cd5cfa7629d6ad3f61b79e/urllib3-1.24.1-py2.py3-none-any.whl
Collecting certifi>=2017.4.17 (from requests->vistir[spinner]>0.2.0->pythonfinder)
  Using cached https://files.pythonhosted.org/packages/56/9d/1d02dd80bc4cd955f98980f28c5ee2200e1209292d5f9e9cc8d030d18655/certifi-2018.10.15-py2.py3-none-any.whl
Installing collected packages: attrs, cached-property, click, colorama, crayons, pyparsing, six, packaging, idna, chardet, urllib3, certifi, requests, cursor, yaspin, vistir, pythonfinder
  Running setup.py develop for pythonfinder
Successfully installed attrs-18.2.0 cached-property-1.5.1 certifi-2018.10.15 chardet-3.0.4 click-7.0 colorama-0.4.0 crayons-0.1.2 cursor-1.2.0 idna-2.7 packaging-18.0 pyparsing-2.3.0 pythonfinder requests-2.20.1 six-1.11.0 urllib3-1.24.1 vistir-0.2.4 yaspin-0.14.0

Adding pythonfinder to Pipfile's [packages]…
Pipfile.lock (ade7d5) out of date, updating to (a65489)…
Locking [dev-packages] dependencies…
Locking [packages] dependencies…
Updated Pipfile.lock (ade7d5)!
Installing dependencies from Pipfile.lock (ade7d5)…
Ignoring colorama: markers 'extra == "spinner"' don't match your environment
Ignoring cursor: markers 'extra == "spinner"' don't match your environment
Ignoring yaspin: markers 'extra == "spinner"' don't match your environment
  🐍   ▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉ 17/17 — 00:00:03
An error occurred while installing -e git+https://github.com/sarugaku/pythonfinder.git@feature/asdf-support@58019b791fc46d464277108180ab6a7b83b8ea8a#egg=pythonfinder! Will try again.
Installing initially failed dependencies…
Obtaining pythonfinder from git+https://github.com/sarugaku/pythonfinder.git@feature/asdf-support@58019b791fc46d464277108180ab6a7b83b8ea8a#egg=pythonfinder
  Cloning https://github.com/sarugaku/pythonfinder.git@feature/asdf-support (to revision 58019b791fc46d464277108180ab6a7b83b8ea8a) to /Users/johria/.local/share/virtualenvs/newproject-pYEZLETE/src/pythonfinder
remote: Not Found
fatal: repository 'https://github.com/sarugaku/pythonfinder.git@feature/asdf-support/' not found

Command "git clone -q https://github.com/sarugaku/pythonfinder.git@feature/asdf-support /Users/johria/.local/share/virtualenvs/newproject-pYEZLETE/src/pythonfinder" failed with error code 128 in None

  ☤  ▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉ 0/1 — 00:00:01
```

-------------------------------------------------------------------------------

Please run `$ pipenv --support`, and paste the results here. Don't put backticks (`` ` ``) around it! The output already contains Markdown formatting.

<details><summary>$ pipenv --support</summary>

Pipenv version: `'2018.10.13'`

Pipenv location: `'/usr/local/Cellar/pipenv/2018.10.13_1/libexec/lib/python3.7/site-packages/pipenv'`

Python location: `'/usr/local/Cellar/pipenv/2018.10.13_1/libexec/bin/python3.7'`

Python installations found:

  - `3.7.1`: `/Users/johria/.asdf/shims/python3`
  - `3.7.1`: `/Users/johria/.asdf/shims/python`
  - `3.7.1`: `/Users/johria/.asdf/shims/python3.7m`
  - `3.7.1`: `/Users/johria/.asdf/shims/python3.7`
  - `3.7.1`: `/usr/local/Cellar/python/3.7.1/Frameworks/Python.framework/Versions/3.7/bin/python3.7`
  - `3.7.1`: `/usr/local/Cellar/python/3.7.1/Frameworks/Python.framework/Versions/3.7/bin/python3.7m`
  - `2.7.15`: `/Users/johria/.asdf/shims/python2`
  - `2.7.15`: `/Users/johria/.asdf/shims/python2.7`
  - `2.7.15`: `/usr/local/Cellar/python@2/2.7.15_1/Frameworks/Python.framework/Versions/2.7/bin/python2.7`
  - `2.7.10`: `/usr/bin/python`
  - `2.7.10`: `/usr/bin/pythonw`
  - `2.7.10`: `/System/Library/Frameworks/Python.framework/Versions/2.7/bin/python2.7`

PEP 508 Information:

```
{'implementation_name': 'cpython',
 'implementation_version': '3.7.1',
 'os_name': 'posix',
 'platform_machine': 'x86_64',
 'platform_python_implementation': 'CPython',
 'platform_release': '17.7.0',
 'platform_system': 'Darwin',
 'platform_version': 'Darwin Kernel Version 17.7.0: Wed Oct 10 23:06:14 PDT '
                     '2018; root:xnu-4570.71.13~1/RELEASE_X86_64',
 'python_full_version': '3.7.1',
 'python_version': '3.7',
 'sys_platform': 'darwin'}
```

System environment variables:

  - `PATH`
  - `RSTUDIO_WHICH_R`
  - `LESS_TERMCAP_mb`
  - `TERM_PROGRAM`
  - `LESS_TERMCAP_md`
  - `LESS_TERMCAP_me`
  - `ANDROID_HOME`
  - `ASDF_DIR`
  - `TERM`
  - `SHELL`
  - `CLICOLOR`
  - `MAKEFLAGS`
  - `SAVEHIST`
  - `HISTSIZE`
  - `TMPDIR`
  - `Apple_PubSub_Socket_Render`
  - `TERM_PROGRAM_VERSION`
  - `FPATH`
  - `LESS_TERMCAP_ue`
  - `TERM_SESSION_ID`
  - `USER`
  - `OSX`
  - `COMMAND_MODE`
  - `SSH_AUTH_SOCK`
  - `__CF_USER_TEXT_ENCODING`
  - `LSCOLORS`
  - `LESS_TERMCAP_us`
  - `NLTK_DATA`
  - `PIPENV_DEFAULT_PYTHON_VERSION`
  - `_`
  - `WORDCHARS`
  - `LESS_TERMEND`
  - `PWD`
  - `JAVA_HOME`
  - `AWS_SECRET_ACCESS_KEY`
  - `EDITOR`
  - `LANG`
  - `ITERM_PROFILE`
  - `XPC_FLAGS`
  - `CPUCOUNT`
  - `XPC_SERVICE_NAME`
  - `AWS_ACCESS_KEY_ID`
  - `SHLVL`
  - `HOME`
  - `COLORFGBG`
  - `GREP_OPTIONS`
  - `BUNDLE_JOBS`
  - `HOMEBREW_PREFIX`
  - `ITERM_SESSION_ID`
  - `LOGNAME`
  - `LESS_TERMCAP_so`
  - `UNIX`
  - `HOMEBREW_CASK_OPTS`
  - `GOPATH`
  - `R_HOME`
  - `DISPLAY`
  - `GIT_PAGER`
  - `SECURITYSESSIONID`
  - `HISTFILE`
  - `LESS_TERMCAP_se`
  - `COLORTERM`
  - `PYTHONDONTWRITEBYTECODE`
  - `PIP_SHIMS_BASE_MODULE`
  - `PIP_PYTHON_PATH`

Pipenv–specific environment variables:

 - `PIPENV_DEFAULT_PYTHON_VERSION`: `3.6`

Debug–specific environment variables:

  - `PATH`: `/usr/local/Cellar/pipenv/2018.10.13_1/libexec/tools:bin:/Users/johria/.asdf/shims:/Users/johria/.asdf/bin:/usr/local/sbin:/usr/local/bin:/Users/johria/bin:/usr/bin:/bin:/usr/sbin:/sbin:/opt/X11/bin:/Users/johria/.antigen/bundles/zdharma/fast-syntax-highlighting:/usr/local/opt/fzf/bin`
  - `SHELL`: `/usr/local/bin/zsh`
  - `EDITOR`: `vim`
  - `LANG`: `en_US.UTF-8`
  - `PWD`: `/Users/johria/Development/newproject`

---------------------------

Contents of `Pipfile` ('/Users/johria/Development/newproject/Pipfile'):

```toml
[[source]]
url = "https://pypi.org/simple"
verify_ssl = true
name = "pypi"

[packages]
pythonfinder = {editable = true, git = "https://github.com/sarugaku/pythonfinder.git@feature/asdf-support"}

[dev-packages]

[requires]
python_version = "3.7"

```

Contents of `Pipfile.lock` ('/Users/johria/Development/newproject/Pipfile.lock'):

```json
{
    "_meta": {
        "hash": {
            "sha256": "4c06403481352f69ac3fd5f1481dfa52d4de559822fc450234f20428efade7d5"
        },
        "pipfile-spec": 6,
        "requires": {
            "python_version": "3.7"
        },
        "sources": [
            {
                "name": "pypi",
                "url": "https://pypi.org/simple",
                "verify_ssl": true
            }
        ]
    },
    "default": {
        "attrs": {
            "hashes": [
                "sha256:10cbf6e27dbce8c30807caf056c8eb50917e0eaafe86347671b57254006c3e69",
                "sha256:ca4be454458f9dec299268d472aaa5a11f67a4ff70093396e1ceae9c76cf4bbb"
            ],
            "version": "==18.2.0"
        },
        "cached-property": {
            "hashes": [
                "sha256:3a026f1a54135677e7da5ce819b0c690f156f37976f3e30c5430740725203d7f",
                "sha256:9217a59f14a5682da7c4b8829deadbfc194ac22e9908ccf7c8820234e80a1504"
            ],
            "version": "==1.5.1"
        },
        "certifi": {
            "hashes": [
                "sha256:339dc09518b07e2fa7eda5450740925974815557727d6bd35d319c1524a04a4c",
                "sha256:6d58c986d22b038c8c0df30d639f23a3e6d172a05c3583e766f4c0b785c0986a"
            ],
            "version": "==2018.10.15"
        },
        "chardet": {
            "hashes": [
                "sha256:84ab92ed1c4d4f16916e05906b6b75a6c0fb5db821cc65e70cbd64a3e2a5eaae",
                "sha256:fc323ffcaeaed0e0a02bf4d117757b98aed530d9ed4531e3e15460124c106691"
            ],
            "version": "==3.0.4"
        },
        "click": {
            "hashes": [
                "sha256:2335065e6395b9e67ca716de5f7526736bfa6ceead690adf616d925bdc622b13",
                "sha256:5b94b49521f6456670fdb30cd82a4eca9412788a93fa6dd6df72c94d5a8ff2d7"
            ],
            "version": "==7.0"
        },
        "colorama": {
            "hashes": [
                "sha256:a3d89af5db9e9806a779a50296b5fdb466e281147c2c235e8225ecc6dbf7bbf3",
                "sha256:c9b54bebe91a6a803e0772c8561d53f2926bfeb17cd141fbabcb08424086595c"
            ],
            "markers": "extra == 'spinner'",
            "version": "==0.4.0"
        },
        "crayons": {
            "hashes": [
                "sha256:5e17691605e564d63482067eb6327d01a584bbaf870beffd4456a3391bd8809d",
                "sha256:6f51241d0c4faec1c04c1c0ac6a68f1d66a4655476ce1570b3f37e5166a599cc"
            ],
            "version": "==0.1.2"
        },
        "cursor": {
            "hashes": [
                "sha256:8ee9fe5b925e1001f6ae6c017e93682583d2b4d1ef7130a26cfcdf1651c0032c"
            ],
            "markers": "extra == 'spinner'",
            "version": "==1.2.0"
        },
        "idna": {
            "hashes": [
                "sha256:156a6814fb5ac1fc6850fb002e0852d56c0c8d2531923a51032d1b70760e186e",
                "sha256:684a38a6f903c1d71d6d5fac066b58d7768af4de2b832e426ec79c30daa94a16"
            ],
            "version": "==2.7"
        },
        "packaging": {
            "hashes": [
                "sha256:0886227f54515e592aaa2e5a553332c73962917f2831f1b0f9b9f4380a4b9807",
                "sha256:f95a1e147590f204328170981833854229bb2912ac3d5f89e2a8ccd2834800c9"
            ],
            "version": "==18.0"
        },
        "pyparsing": {
            "hashes": [
                "sha256:40856e74d4987de5d01761a22d1621ae1c7f8774585acae358aa5c5936c6c90b",
                "sha256:f353aab21fd474459d97b709e527b5571314ee5f067441dc9f88e33eecd96592"
            ],
            "version": "==2.3.0"
        },
        "pythonfinder": {
            "editable": true,
            "git": "https://github.com/sarugaku/pythonfinder.git@feature/asdf-support",
            "ref": "58019b791fc46d464277108180ab6a7b83b8ea8a"
        },
        "requests": {
            "hashes": [
                "sha256:65b3a120e4329e33c9889db89c80976c5272f56ea92d3e74da8a463992e3ff54",
                "sha256:ea881206e59f41dbd0bd445437d792e43906703fff75ca8ff43ccdb11f33f263"
            ],
            "version": "==2.20.1"
        },
        "six": {
            "hashes": [
                "sha256:70e8a77beed4562e7f14fe23a786b54f6296e34344c23bc42f07b15018ff98e9",
                "sha256:832dc0e10feb1aa2c68dcc57dbb658f1c7e65b9b61af69048abc87a2db00a0eb"
            ],
            "version": "==1.11.0"
        },
        "urllib3": {
            "hashes": [
                "sha256:61bf29cada3fc2fbefad4fdf059ea4bd1b4a86d2b6d15e1c7c0b582b9752fe39",
                "sha256:de9529817c93f27c8ccbfead6985011db27bd0ddfcdb2d86f3f663385c6a9c22"
            ],
            "version": "==1.24.1"
        },
        "vistir": {
            "extras": [
                "spinner"
            ],
            "hashes": [
                "sha256:851bd783f2b85a372e563db741dc689cb9263ce2e067e387facdca0c36b6a6ea",
                "sha256:b38ffc8ef83f85d81b4efa4cd31ea3bcd37bdb2bc9e8da9f20a40859bc44b57e"
            ],
            "version": "==0.2.4"
        },
        "yaspin": {
            "hashes": [
                "sha256:36fdccc5e0637b5baa8892fe2c3d927782df7d504e9020f40eb2c1502518aa5a",
                "sha256:8e52bf8079a48e2a53f3dfeec9e04addb900c101d1591c85df69cf677d3237e7"
            ],
            "markers": "extra == 'spinner'",
            "version": "==0.14.0"
        }
    },
    "develop": {}
}

```
</details>

### PR DESCRIPTION
**Pull request by techalchemy**
Update pythonfinder, requirementslib and vistir

- Fixes #3214

### CODE REVIEW

File: pipenv/vendor/vendor.txt [hunk 1]
```diff
@@ -21,20 +21,20 @@ pipdeptree==0.13.0
...
-requirementslib==1.2.5
+requirementslib==1.3.0
     attrs==18.2.0
     distlib==0.2.8
     packaging==18.0
     pyparsing==2.2.2
     pytoml==0.1.19
     plette==0.2.2
-        tomlkit==0.4.6
+        tomlkit==0.5.2
```
Comment (frostming) [inline pipenv/vendor/vendor.txt]: Why is this line here? Vendor update will fail.

Comment (techalchemy) [reply pipenv/vendor/vendor.txt]: it just got caught in some requirementslib updates or whatever, doesn't matter much if that fails anyway i'll just override it if needed

Comment (techalchemy) [reply pipenv/vendor/vendor.txt]: lets just pray that everything else works

**Comment by frostming** [comment]
@techalchemy I am wondering if https://github.com/pypa/pipenv/blob/master/pipenv/test_script.py is added by accident. Maybe you can remove it here.

**Comment by techalchemy** [comment]
Lol. I must have pulled that out of a git stash :) I was trying clicks filesystem isolation fo see if it’s faster than our test approach. It isn’t.
